src/analysis/normed_space/spectrum.lean

```diff
@@ -0,0 +1,182 @@
+/-
+Copyright (c) 2021 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import algebra.algebra.spectrum
+import analysis.calculus.deriv
+/-!
+# The spectrum of elements in complete normed algebra
+
+This file contains the basic theory for the resolvent and spectrum of a Banach algebra.
+
+## Main definitions
+
+* `spectral_radius`: supremum of `abs k` for all `k ∈ spectrum 𝕜 a`
+
+## Main statements
+
+* `is_open_resolvent_set`: the resolvent set is open.
+* `is_closed`: the spectrum is closed.
+* `subset_closed_ball_norm`: the spectrum is a subset of closed disk of radius equal to the norm.
+* `is_compact`: the spectrum is compact.
+* `spectral_radius_le_norm`: the spectral radius is bounded above by the norm.
+* `resolvent_has_deriv_at`: the resolvent function is differentiable on the resolvent set.
+
+
+## TODO
+
+* after we have Liouville's theorem, prove that the spectrum is nonempty when the
+  scalar field is ℂ.
+* compute all derivatives of `resolvent a`.
+
+-/
+
+/-- The *spectral radius* is the supremum of the norm of elements in the spectrum.
+    The spectrum is compact, and so when it is nonempty, this supremum can be
+    achieved. Note that it is possible for `spectrum 𝕜 a = ∅`.  In this case,
+    `spectral_radius a = 0` by the definition of `Sup` on `ℝ`.-/
+noncomputable def spectral_radius (𝕜 : Type*) {A : Type*} [normed_field 𝕜] [ring A]
+  [algebra 𝕜 A] (a : A) : ℝ :=
+Sup (norm '' (spectrum 𝕜 a))
```

Comment:
```suggestion
⨆ k ∈ spectrum 𝕜 a, ∥k∥
```
This means more or less the same thing, but the notation is nice, and there are some convenience lemmas available for it (keywords `supr` or perhaps `bsupr`)

Reply:
changed, and defined it as an `ennreal` instead so that `supr` makes more sense.